walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java

```diff
@@ -42,39 +42,56 @@ public int compare(TimeRange a, TimeRange b) {
   public static final Comparator<TimeRange> ORDER_BY_END = new Comparator<TimeRange>() {
     @Override
     public int compare(TimeRange a, TimeRange b) {
-      return Long.compare(a.end(), b.end());
+      return Long.compare(a.getEnd(), b.getEnd());
     }
   };
 
-  private final int start;
-  private final int duration;
+  private int start;
+  private int duration;
 
   private TimeRange(int start, int duration) {
     this.start = start;
     this.duration = duration;
   }
 
+  public TimeRange() {
+      this.start = 0;
+      this.duration = 0;
+  }
+
   /**
    * Returns the start of the range in minutes.
    */
-  public int start() {
+  public int getStart() {
     return start;
   }
 
+  public void setStart(int start) {
+    this.start = start;
+  }
+
   /**
    * Returns the number of minutes between the start and end.
    */
-  public int duration() {
+  public int getDuration() {
     return duration;
   }
 
+  public void setDuration(int duration) {
+    this.duration = duration;
+  }
+
   /**
    * Returns the end of the range. This ending value is the closing exclusive bound.
    */
-  public int end() {
+  public int getEnd() {
     return start + duration;
   }
 
+  public boolean getValidRange() {
```

Comment:
Since this isn't returning a "valid range" object / concept, but rather answering the question: is this range itself valid it'd be better to name this: "isValid". Eg this isn't a "getter" so to speak

I'd also be careful with "valid" as it really raises the question: what is invalid? To me an invalid range is in a bad state. (Say a negative duration, or a time that doesn't exist.) It's best to design things so that's impossible to phrase, but isn't always possible. Reading the code above, it's not super clear to me what "valid"/"invalid" mean, or how things get in that state.

Reply:
changing to hasPositiveDuration